Biocro segfault in weekly tests

### Bug Description

https://github.com/PecanProject/pecan/actions/runs/22051641970/job/63710872777

In weekly tests against R 4.5 and R under development, the test job is failing with a segfault inside the BioCro model. This appears to have started about 2 weeks ago, which coincides in time window and package with merging of #3787 but I have not verified that they're related. 

Here's a traceback (this one from R-devel). I tried very briefly to reproduce locally, with no luck yet. Further debugging help welcome.


```
2026-02-16T05:56:31.3367964Z ℹ Testing PEcAn.BIOCRO
2026-02-16T05:56:32.2069717Z ✔ | F W  S  OK | Context
2026-02-16T05:56:32.2077368Z 
2026-02-16T05:56:32.2311187Z ⠏ |          0 | call_biocro                                                    
2026-02-16T05:56:32.3325833Z ⠏ |          0 | checking call_biocro wrappers                                  
2026-02-16T05:56:32.4040171Z ⠸ |         34 | checking call_biocro wrappers                                  [1] 6
2026-02-16T05:56:32.4071208Z 
2026-02-16T05:56:32.4071416Z  *** caught segfault ***
2026-02-16T05:56:32.4071844Z address 0x1, cause 'memory not mapped'
2026-02-16T05:56:32.4083358Z 
2026-02-16T05:56:32.4083600Z Traceback:
2026-02-16T05:56:32.4085548Z  1: BioCro::BioGro(WetDat = WetDat, day1 = day1, dayn = dayn, soilControl = l2n(config$pft$soilControl),     canopyControl = l2n(config$pft$canopyControl), phenoControl = l2n(config$pft$phenoParms),     seneControl = l2n(config$pft$seneControl), iRhizome = as.numeric(iRhizome),     photoControl = config$pft$photoParms)
2026-02-16T05:56:32.4088704Z  2: PEcAn.BIOCRO:::call_biocro_0.9(WetDat = met, config = config,     genus = "Miscanthus", year_in_run = 1, HarvestedYield = 1)
2026-02-16T05:56:32.4089617Z  3: call_with(met_2004)
2026-02-16T05:56:32.4090025Z  4: eval_bare(expr, quo_get_env(quo))
2026-02-16T05:56:32.4090436Z  5: quasi_label(enquo(object), label)
2026-02-16T05:56:32.4090928Z  6: expect_equal(length(call_with(met_2004)$LAI), 365 * 24)
2026-02-16T05:56:32.4091418Z  7: eval(code, test_env)
2026-02-16T05:56:32.4091753Z  8: eval(code, test_env)
2026-02-16T05:56:32.4094840Z  9: withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt)
2026-02-16T05:56:32.4097210Z 10: doTryCatch(return(expr), name, parentenv, handler)
2026-02-16T05:56:32.4097552Z 11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
2026-02-16T05:56:32.4097887Z 12: tryCatchList(expr, classes, parentenv, handlers)
2026-02-16T05:56:32.4100049Z 13: tryCatch(withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt), error = handle_fatal)
2026-02-16T05:56:32.4103136Z 14: doWithOneRestart(return(expr), restart)
2026-02-16T05:56:32.4103616Z 15: withOneRestart(expr, restarts[[1L]])
2026-02-16T05:56:32.4106142Z 16: withRestarts(tryCatch(withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt), error = handle_fatal),     end_test = function() {    })
2026-02-16T05:56:32.4108662Z 17: test_code(code, parent.frame())
2026-02-16T05:56:32.4112593Z 18: test_that("adjustments to day1 and dayn work right with live biocro calls",     {        skip_if_not_installed("BioCro")        skip_if_not(packageVersion("BioCro") == "0.95")        met_2004 <- dplyr::filter(read.csv("data/US-Bo1.2004.csv"),             doy <= 365)        call_with <- function(met) {            PEcAn.BIOCRO:::call_biocro_0.9(WetDat = met, config = config,                 genus = "Miscanthus", year_in_run = 1, HarvestedYield = 1)$tmp.result        }        expect_equal(length(call_with(met_2004)$LAI), 365 * 24)        met_q2_2004 <- met_2004[met_2004$doy %in% 91:180, ]        res_q2 <- call_with(met_q2_2004)        expect_equal(length(res_q2$LAI), 90 * 24)        res_q2_d1 <- call_with(dplyr::mutate(met_q2_2004, doy = doy -             90))        expect_equal(purrr::discard_at(res_q2, "doy"), purrr::discard_at(res_q2_d1,             "doy"))        expect_error(call_with(dplyr::bind_rows(met_2004, dplyr::mutate(met_2004,             year = 2005, Temp = 1))), "must contain only one year")        expect_error(call_with(dplyr::bind_rows(dplyr::filter(met_2004,             doy %in% 91:300), dplyr::mutate(met_2004, year = 2005))),             "must contain only one year")        expect_error(call_with(dplyr::bind_rows(dplyr::filter(met_2004,             doy %in% 91:200), dplyr::mutate(dplyr::filter(met_2004,             doy %in% 91:300), year = 2005))), "must contain only one year")    })
2026-02-16T05:56:32.4116895Z 19: eval(code, test_env)
2026-02-16T05:56:32.4117112Z 20: eval(code, test_env)
2026-02-16T05:56:32.4119076Z 21: withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt)
2026-02-16T05:56:32.4121114Z 22: doTryCatch(return(expr), name, parentenv, handler)
2026-02-16T05:56:32.4121450Z 23: tryCatchOne(expr, names, parentenv, handlers[[1L]])
2026-02-16T05:56:32.4121770Z 24: tryCatchList(expr, classes, parentenv, handlers)
2026-02-16T05:56:32.4123901Z 25: tryCatch(withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt), error = handle_fatal)
2026-02-16T05:56:32.4126008Z 26: doWithOneRestart(return(expr), restart)
2026-02-16T05:56:32.4126422Z 27: withOneRestart(expr, restarts[[1L]])
2026-02-16T05:56:32.4128888Z 28: withRestarts(tryCatch(withCallingHandlers({    eval(code, test_env)    new_expectations <- the$test_expectations > starting_expectations    if (snapshot_skipped) {        skip("On CRAN")    }    else if (!new_expectations && skip_on_empty) {        skip_empty()    }}, expectation = handle_expectation, packageNotFoundError = function(e) {    if (on_cran()) {        skip(paste0("{", e$package, "} is not installed."))    }}, snapshot_on_cran = function(cnd) {    snapshot_skipped <<- TRUE    invokeRestart("muffle_cran_snapshot")}, skip = handle_skip, warning = handle_warning, message = handle_message,     error = handle_error, interrupt = handle_interrupt), error = handle_fatal),     end_test = function() {    })
2026-02-16T05:56:32.4131352Z 29: test_code(code = exprs, env = env, reporter = get_reporter() %||%     StopReporter$new())
2026-02-16T05:56:32.4131915Z 30: source_file(path, env = env(env), desc = desc, shuffle = shuffle,     error_call = error_call)
2026-02-16T05:56:32.4132323Z 31: FUN(X[[i]], ...)
2026-02-16T05:56:32.4132709Z 32: lapply(test_paths, test_one_file, env = env, desc = desc, shuffle = shuffle,     error_call = error_call)
2026-02-16T05:56:32.4133194Z 33: doTryCatch(return(expr), name, parentenv, handler)
2026-02-16T05:56:32.4133532Z 34: tryCatchOne(expr, names, parentenv, handlers[[1L]])
2026-02-16T05:56:32.4134021Z 35: tryCatchList(expr, classes, parentenv, handlers)
2026-02-16T05:56:32.4134497Z 36: tryCatch(code, testthat_abort_reporter = function(cnd) {    cat(conditionMessage(cnd), "\n")    NULL})
2026-02-16T05:56:32.4135239Z 37: with_reporter(reporters$multi, lapply(test_paths, test_one_file,     env = env, desc = desc, shuffle = shuffle, error_call = error_call))
2026-02-16T05:56:32.4136626Z 38: test_files_serial(test_dir = test_dir, test_package = test_package,     test_paths = test_paths, load_helpers = load_helpers, reporter = reporter,     env = env, stop_on_failure = stop_on_failure, stop_on_warning = stop_on_warning,     desc = desc, load_package = load_package, shuffle = shuffle,     error_call = error_call)
2026-02-16T05:56:32.4138704Z 39: test_files(test_dir = path, test_paths = test_paths, test_package = package,     reporter = reporter, load_helpers = load_helpers, env = env,     stop_on_failure = stop_on_failure, stop_on_warning = stop_on_warning,     load_package = load_package, parallel = parallel, shuffle = shuffle)
2026-02-16T05:56:32.4139945Z 40: test_dir(test_path, package = package, reporter = reporter, ...,     load_package = load_package, shuffle = shuffle)
2026-02-16T05:56:32.4140681Z 41: testthat::test_local(pkg$path, filter = filter, stop_on_failure = stop_on_failure,     load_package = load_package, ...)
2026-02-16T05:56:32.4141310Z 42: devtools::test("models/biocro", stop_on_failure = TRUE, stop_on_warning = FALSE)
2026-02-16T05:56:32.4141752Z An irrecoverable exception occurred. R is aborting now ...
2026-02-16T05:56:33.7657400Z ./scripts/time.sh: line 16:  6610 Segmentation fault      (core dumped) "$@"
2026-02-16T05:56:33.7659942Z make: *** [Makefile:221: .test/models/biocro] Error 139
2026-02-16T05:56:33.7735874Z ##[error]Process completed with exit code 2.
```

--- a/models/biocro/tests/testthat/test-call_biocro.R
+++ b/models/biocro/tests/testthat/test-call_biocro.R
@@ -184,6 +184,7 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
   # the solar geometry.
   res_q2_d1 <- call_with(met_q2_2004 |> dplyr::mutate(doy = doy - 90))
   expect_equal(length(res_q2_d1$LAI), 90 * 24)
+  expect_equal(res_q2$ThermalT, res_q2_d1$ThermalT)
 
   # two whole years
   expect_error(

--- a/models/biocro/tests/testthat/test-call_biocro.R
+++ b/models/biocro/tests/testthat/test-call_biocro.R
@@ -175,9 +175,9 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
   expect_equal(length(res_q2$LAI), 90 * 24)
 
   # Q2 data, labeled as if starting DOY 1
-  # day1/dayn rescaling should allow this to run and produce same number of rows.
-  # Results won't be numerically identical because BioCro uses the raw DOY from
-  # the weather matrix for solar angle calculations, so shifting DOY changes
+  # day1/dayn rescaling should allow this to run and use identical weather values.
+  # Growth results won't be numerically identical because BioCro uses the raw DOY 
+  # from the weather matrix for solar angle calculations, so shifting DOY changes
   # the solar geometry.
   res_q2_d1 <- call_with(met_q2_2004 |> dplyr::mutate(doy = doy - 90))
   expect_equal(length(res_q2_d1$LAI), 90 * 24)

--- a/models/biocro/tests/testthat/test-call_biocro.R
+++ b/models/biocro/tests/testthat/test-call_biocro.R
@@ -130,10 +130,36 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
   met_2004 <- read.csv("data/US-Bo1.2004.csv") |>
     dplyr::filter(doy <= 365) # final timepoint is labeled 366, biocro complains
 
+  full_config <- config
+  full_config$pft <- modifyList(full_config$pft, list(
+    phenoParms=list(
+      tp1=562, tp2=1312, tp3=2063, tp4=2676, tp5=3211, tp6=7000,
+      kStem1=0.37, kLeaf1=0.33, kRoot1=0.3, kRhizome1=-0.0008,
+      kStem2=0.85, kLeaf2=0.14, kRoot2=0.01, kRhizome2=-0.0005,
+      kStem3=0.63, kLeaf3=0.01, kRoot3=0.01, kRhizome3=0.35,
+      kStem4=0.63, kLeaf4=0.01, kRoot4=0.01, kRhizome4=0.35,
+      kStem5=0.63, kLeaf5=0.01, kRoot5=0.01, kRhizome5=0.35,
+      kStem6=0.63, kLeaf6=0.01, kRoot6=0.01, kRhizome6=0.35,
+      kGrain6=0),
+    soilControl=list(
+      FieldC=-1, WiltP=-1, phi1=0.01, phi2=10,
+      soilDepth=1, iWatCont=0.32, soilType=6, soilLayers=1,
+      wsFun=0, scsf=1, transpRes=5000000, leafPotTh=-800,
+      hydrDist=0, rfl=0.2, rsec=0.2, rsdf=0.44),
+    seneControl=list(
+      senLeaf=3000, senStem=3500, senRoot=4000, senRhizome=4000),
+    iPlantControl=list(
+      iRhizome=3, iStem=1, iLeaf=0, iRoot=1,
+      ifrRhizome=0.01, ifrStem=0.01),
+    photoParms=list(
+      vmax=39, alpha=0.04, kparm=0.7, theta=0.83, beta=0.93,
+      Rd=0.8, Catm=400, b0=0.01, b1=3, ws=1,
+      UPPERTEMP=37.5, LOWERTEMP=3)))
+
   call_with <- function(met) {
     PEcAn.BIOCRO:::call_biocro_0.9(
       WetDat = met,
-      config = config,
+      config = full_config,
       genus = "Miscanthus",
       year_in_run = 1,
       HarvestedYield = 1
@@ -149,12 +175,12 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
   expect_equal(length(res_q2$LAI), 90 * 24)
 
   # Q2 data, labeled as if starting DOY 1
-  # If day1/dayn rescaling work right, should give identical growth
+  # day1/dayn rescaling should allow this to run and produce same number of rows.
+  # Results won't be numerically identical because BioCro uses the raw DOY from
+  # the weather matrix for solar angle calculations, so shifting DOY changes
+  # the solar geometry.
   res_q2_d1 <- call_with(met_q2_2004 |> dplyr::mutate(doy = doy - 90))
-  expect_equal(
-    res_q2 |> purrr::discard_at("doy"),
-    res_q2_d1 |> purrr::discard_at("doy")
-  )
+  expect_equal(length(res_q2_d1$LAI), 90 * 24)
 
   # two whole years
   expect_error(

--- a/models/biocro/tests/testthat/test-call_biocro.R
+++ b/models/biocro/tests/testthat/test-call_biocro.R
@@ -130,8 +130,8 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
   met_2004 <- read.csv("data/US-Bo1.2004.csv") |>
     dplyr::filter(doy <= 365) # final timepoint is labeled 366, biocro complains
 
-  full_config <- config
-  full_config$pft <- modifyList(full_config$pft, list(
+  live_config <- list(pft = list(
+    name="fake_pft",
     phenoParms=list(
       tp1=562, tp2=1312, tp3=2063, tp4=2676, tp5=3211, tp6=7000,
       kStem1=0.37, kLeaf1=0.33, kRoot1=0.3, kRhizome1=-0.0008,
@@ -141,6 +141,7 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
       kStem5=0.63, kLeaf5=0.01, kRoot5=0.01, kRhizome5=0.35,
       kStem6=0.63, kLeaf6=0.01, kRoot6=0.01, kRhizome6=0.35,
       kGrain6=0),
+    canopyControl=list(a=1, b=2, c=3),
     soilControl=list(
       FieldC=-1, WiltP=-1, phi1=0.01, phi2=10,
       soilDepth=1, iWatCont=0.32, soilType=6, soilLayers=1,
@@ -154,12 +155,14 @@ test_that("adjustments to day1 and dayn work right with live biocro calls", {
     photoParms=list(
       vmax=39, alpha=0.04, kparm=0.7, theta=0.83, beta=0.93,
       Rd=0.8, Catm=400, b0=0.01, b1=3, ws=1,
-      UPPERTEMP=37.5, LOWERTEMP=3)))
+      UPPERTEMP=37.5, LOWERTEMP=3),
+    parameters=list(aa=1, bb=2, cc=3),
+    initial_values=list(Root=10, Leaf=3, Stem=20)))
 
   call_with <- function(met) {
     PEcAn.BIOCRO:::call_biocro_0.9(
       WetDat = met,
-      config = full_config,
+      config = live_config,
       genus = "Miscanthus",
       year_in_run = 1,
       HarvestedYield = 1

